src/components/app/load-data.test.js

```diff
@@ -0,0 +1,74 @@
+import { getInitialState, loadData } from './load-data';
+import { saveState } from '../../utils';
+import formatData from '../../utils/format-data';
+import loremIpsum from '../../utils/data/lorem-ipsum.mock';
+import animals from '../../utils/data/animals.mock';
+import demo from '../../utils/data/demo.mock';
+
+describe('load-data', () => {
+  describe('getInitialState', () => {
+    const loremData = loadData('lorem');
+
+    it('returns an object', () => {
+      expect(getInitialState(loremData)).toEqual(expect.any(Object));
+    });
+
+    it('does not require the second argument', () => {
+      expect(getInitialState(loremData)).toEqual(
+        getInitialState(loremData, {})
+      );
+    });
+
+    it('returns full initial state', () => {
+      expect(getInitialState(loremData)).toMatchObject({
+        ...loremData,
+        chartSize: {},
+        parameters: true,
+        textLabels: true,
+        theme: 'dark',
+        view: 'combined',
+        visible: { labelBtn: true, themeBtn: true }
+      });
+    });
+
+    it('returns mutated visible state if prop is set', () => {
+      const visible = { themeBtn: false };
+      expect(getInitialState(loremData, { visible })).toMatchObject({
+        visible: { labelBtn: true, themeBtn: false }
+      });
+    });
+
+    it('uses prop values instead of defaults if provided', () => {
+      expect(getInitialState(loremData, { theme: 'light' })).toMatchObject({
+        theme: 'light'
+      });
+    });
```

Comment:
Could this test and the one above be merged i.e. are they testing the same thing (mutation of expected state) e.g.

```
it('returns mutated state if prop is set', () => {
      const mutate = { visible: { themeBtn: false }, theme: 'light' };
      expect(getInitialState(loremData, mutate)).toMatchObject({
        visible: { labelBtn: true, themeBtn: false },
        theme: 'light'
      });
    });
```

Reply:
👍